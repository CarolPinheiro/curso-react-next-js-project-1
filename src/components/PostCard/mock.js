export const postCardPropsMock = {
  title: "any-title",
  body: "any-body",
  id: 1,
  cover: "any-image.png",
};
