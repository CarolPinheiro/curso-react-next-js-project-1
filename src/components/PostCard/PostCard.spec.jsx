import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { postCardPropsMock as props } from "./mock";

describe("<PostCard/>", () => {
  it("should render PostCard correctly", () => {
    render(<PostCard {...props} />);

    expect(screen.getByAltText(/any-title/i)).toHaveAttribute(
      "src",
      "any-image.png"
    );

    expect(
      screen.getByRole("heading", { name: /any-title/i })
    ).toBeInTheDocument();

    expect(screen.getByText("any-body")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard {...props} />);

    expect(container).toMatchSnapshot();
  });
});
