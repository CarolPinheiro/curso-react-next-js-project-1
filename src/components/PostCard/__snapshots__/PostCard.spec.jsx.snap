// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div>
  <div
    class="post"
  >
    <img
      alt="any-title"
      src="any-image.png"
    />
    <div
      class="post-content"
    >
      <h2>
        any-title
      </h2>
      <p>
        any-body
      </p>
    </div>
  </div>
</div>
`;
